webpack/assets/javascripts/react_app/components/common/TypeAheadSelect/TypeAheadSelectReducer.js

```diff
@@ -0,0 +1,23 @@
+import Immutable from 'seamless-immutable';
+import { UPDATE_OPTIONS, UPDATE_SELECTED } from './TypeAheadSelectConstants';
+
+const initialState = Immutable({});
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case UPDATE_OPTIONS:
+      return state.setIn([action.payload.id], {
+        ...state[action.payload.id],
+        options: action.payload.options,
+      });
+    case UPDATE_SELECTED:
+      return state.setIn([action.payload.id], {
+        ...state[action.payload.id],
+        selected: action.payload.items.map(item =>
```

Comment:
can you destruct the received action into something like: 
`export default (state = initialState, { type, payload: { id, options, items } = {} } => `

Reply:
IMO this should be `selected: payload.items` while the mapping can be in a memorized selector.

Reply:
Or maybe just 
update items and in a selector return "selected" https://github.com/theforeman/foreman/pull/7387#discussion_r371396978